pubsub_nats_options.go

```diff
@@ -12,15 +12,35 @@
 package bahamut
 
 import (
-	"bytes"
 	"context"
 	"crypto/tls"
 	"fmt"
 	"time"
 
-	nats "github.com/nats-io/go-nats"
+	"github.com/nats-io/go-nats"
 )
 
+type requestMode int
+
+const (
+	requestModeNoReply requestMode = iota
+	requestModeACK
+	requestModePublication
+)
+
+func (m requestMode) String() string {
+	switch m {
+	case requestModeNoReply:
+		return "requestModeNoReply"
+	case requestModeACK:
+		return "requestModeACK"
+	case requestModePublication:
+		return "requestModePublication"
+	default:
+		return "unknown"
+	}
+}
```

Comment:
This is so that when you print the `requestMode` it results in a human friendly representation of the constant as opposed to its literal value.

The `fmt` package looks to see if your type implements the `Stringer` interface and if so it calls your `String` method.